client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/RemoteCacheManagerAdminImpl.java

```diff
@@ -0,0 +1,52 @@
+package org.infinispan.client.hotrod.impl;
+
+import java.util.Collections;
+import java.util.EnumSet;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.StringJoiner;
+
+import org.infinispan.client.hotrod.AdminFlag;
+import org.infinispan.client.hotrod.RemoteCacheManagerAdmin;
+import org.infinispan.client.hotrod.impl.operations.OperationsFactory;
+
+/**
+ * @author Tristan Tarrant
+ * @since 9.0
+ */
+
+public class RemoteCacheManagerAdminImpl implements RemoteCacheManagerAdmin {
+   public static final String CACHE_NAME = "CACHE_NAME";
+   public static final String CACHE_TEMPLATE = "CACHE_TEMPLATE";
+   public static final String FLAGS = "FLAGS";
+   private final OperationsFactory operationsFactory;
+
+   public RemoteCacheManagerAdminImpl(OperationsFactory operationsFactory) {
+      this.operationsFactory = operationsFactory;
+   }
+
+   @Override
+   public void createCache(String name, String template) {
+      createCache(name, template, EnumSet.noneOf(AdminFlag.class));
+   }
+
+   @Override
+   public void createCache(String name, String template, EnumSet<AdminFlag> flags) {
+      Map<String, String> params = new HashMap<>(2);
+      params.put(CACHE_NAME, name);
+      if (template != null) params.put(CACHE_TEMPLATE, template);
+      if (flags != null && !flags.isEmpty()) params.put(FLAGS, flagsToString(flags));
+      operationsFactory.newAdminOperation("CACHE_CREATE", params).execute();
```

Comment:
Not only would it simplify the server, but it'd mean any client (Java, Node.js, C/C++/.Net) could just provide a facade around an execute call with certain parameters.